### ISSUE TITLE
model.predict leaks memory

### ISSUE THREAD

**Issue by sspohrer** [tensorflow/tensorflow#35835]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes. We are using our own custom layer to build a keras model.

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Mac & Ubuntu 16.04. Python 3.7 
- TensorFlow installed from (source or binary):
Using the stable version of tensorflow 2.0.0, but tf 2.1 appears to exhibit the same issue
- TensorFlow version (use command below):
v2.0.0-rc2-26-g64c3d382ca 2.0.0
- Python version:
3.7.6

**Describe the current behavior**
When calling model.predict in a loop memory grows to many gigs. Below is a very simple example of what we're doing that experiences severe memory growth. Granted there are many layers involved, some custom. Each call to predict is growing in memory size.

```
model = tf.keras.Model(sent_input, preds)
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['acc'])
model.load_weights(...)
while True:
    inputs = prepare_inputs(x)
    model.predict(inputs)
```

I have seen some other reports that suggest predict_on_batch might be used, but this also exhibits the same issue.

**Describe the expected behavior**
I would expect to be able to call predict as many times as necessary and for memory consumption to remain mostly constant.

**Other info / logs**
In my debugging I was able to find this using pympler. I was able to attach an image of what I believe is the culprit  object being leaked, but I could be mistaken on this. It looks like EagerTensors are being leaked. It appears the TimeDistributedLayer is keeping references to them.

```
                                                 types |   # objects |   total size
====================================================== | =========== | ============
                                                  dict |          20 |      2.07 KB
    tensorflow.python.framework.tensor_shape.Dimension |          13 |    832     B
           tensorflow.python.framework.ops.EagerTensor |           3 |    528     B
                                                  list |           5 |    520     B
  tensorflow.python.framework.tensor_shape.TensorShape |           5 |    320     B
```

![objgraph-qwasabxu](https://user-images.githubusercontent.com/5382738/72302369-73a86500-362f-11ea-9044-93cd2b1605e2.png)

**Comment by sspohrer** [comment]
The code change suggested here seems to resolve the issue, though I am not sure this is the correct change. https://github.com/tensorflow/tensorflow/issues/33178#issuecomment-549079302

I do not see this change available in any of the pip packages available. Though it sounds like it maybe available on a nightly branch somewhere? 

I noticed that 2.1 has a leak still, so i'm not sure if the change has been pulled in yet?

**Comment by weihnachten033** [comment]
The PR for the issue is this one :  #33441

**Comment by goldiegadde** [comment]
@weihnachten033 @sspohrer this PR has been merged 2 days ago, so it should be availabe in the latest tf-nightly. It is not in TF 2.1.0 and will be available in the next TF release.

**Comment by goldiegadde** [comment]
@sspohrer were you able to test with latest tf-nightly ?

**Comment by Saduf2019** [comment]
@sspohrer could you please respond to the above comment

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Saduf2019** [comment]
@sspohrer 
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
